src/libtimed/models.py

```diff
@@ -43,6 +43,21 @@ def __init__(self, client) -> None:
     relationships: list[tuple]
     filters: list[tuple]
 
+    def _get_included_item(self, item_type: str, item_id, included: dict):
+        included_item = next(
+            (
+                included_item
+                for included_item in included
+                if included_item.get("type") == item_type
+                and included_item.get("id") == item_id
+            ),
+            None,
+        )
+        if not included_item:
+            return None
+        included_model = self.client._type_map[item_type](self.client)
+        return included_model._deserialize(included_item, included)
```

Comment:
`_deserialize()` used to fully contain the recursion logic, so the stack looked something like so:

- `_deserialize()`
  - `_deserialize()`
    -  `_deserialize()`
        - ...

with this change the stack will look more like this:

- `_deserialize()`
  - `_get_included_item()`
    - `_deserialize()`
      - `_get_included_item()`
        - `_deserialize()`
          - `_get_included_item()`
            - ...

is this a trade off, you're willing to make?

Reply:
yes